terraform/gitlab/ci-templates/k8s-cluster/.gitlab/scripts/dictmerge.py

```
@@ -81,10 +81,14 @@ def mergeListOfDicts(data1, data2, fileName, outputFilename, fileType):
 if os.path.isfile(default_config_file):
     if defaultExt == ".yaml":
         with open(default_config_file, 'r') as f:
-            data1 = yaml.load(f, Loader=yaml.SafeLoader)
+            data1 = yaml.load(f, Loader=yaml.SafeLoader) or {}
     elif defaultExt == ".json":
-        with open(default_config_file, 'r') as f:
-            data1 = json.load(f)
+        try:
+            with open(default_config_file, 'r') as f:
+                data1 = json.load(f)
+        except json.JSONDecodeError:
+            print("  Could not parse the default config file", default_config_file," assigning empty dict")
+            data1 = {}
```

As the default comes from IaC, it is best to crash if it cannot be parsed. Proceeding with an empty config can be dangerous or lead to other errors, which will be hard to trace to the silent failure here.